### PR TITLE
test: skip several tests in test-seccomp when running under valgrind or ASAN

### LINKED ISSUE #17630: test: test-seccomp fails under valgrind (and ASAN)
**systemd version the issue has been seen with**
 v247-rc2

**Used distribution**
Fedora 33 x86_64

**Linux kernel version used** (`uname -a`)
Linux thinkpad-x280 5.9.8-200.fc33.x86_64 #1 SMP Tue Nov 10 21:58:19 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

**CPU architecture issue was seen on**
x86_64

**Unexpected behaviour you saw**
```
$ rpm -q valgrind
valgrind-3.16.1-5.fc33.x86_64
$ rpm -q libseccomp
libseccomp-2.5.0-3.fc33.x86_64
```

With the following additional debug log,
```
diff --git a/src/shared/seccomp-util.c b/src/shared/seccomp-util.c
index 6ca2c5713d..ec482ba56f 100644
--- a/src/shared/seccomp-util.c
+++ b/src/shared/seccomp-util.c
@@ -1042,17 +1042,21 @@ int seccomp_load_syscall_filter_set(uint32_t default_action, const SyscallFilter
 
                 r = seccomp_init_for_arch(&seccomp, arch, default_action);
                 if (r < 0)
-                        return r;
+                        return log_debug_errno(r, "Failed to init: %m");
 
+                log_debug("%s: init success", __func__);
                 r = add_syscall_filter_set(seccomp, set, action, NULL, log_missing, NULL);
                 if (r < 0)
                         return log_debug_errno(r, "Failed to add filter set: %m");
 
+                log_debug("%s: add success", __func__);
                 r = seccomp_load(seccomp);
                 if (ERRNO_IS_SECCOMP_FATAL(r))
-                        return r;
+                        return log_debug_errno(r, "Failed to load: %m");
                 if (r < 0)
                         log_debug_errno(r, "Failed to install filter set for architecture %s, skipping: %m", seccomp_arch_to_string(arch));
+
+                log_debug("%s: load success", __func__);
         }
 
         return 0;
```
you can see the follwoing:
```
$ sudo valgrind --leak-check=full ./test-seccomp
(snip)
/* test_filter_sets */
Testing @default
Operating on architecture: x86
--23577-- WARNING: unhandled amd64-linux syscall: 317
--23577-- You may be able to write your own handler.
--23577-- Read the file README_MISSING_SYSCALL_OR_IOCTL.
--23577-- Nevertheless we consider this a bug.  Please report
--23577-- it at http://valgrind.org/support/bug_reports.html.
seccomp_load_syscall_filter_set: init success
Failed to add rule for system call cacheflush() / -10104, ignoring: Numerical argument out of domain
Failed to add rule for system call set_tls() / -10183, ignoring: Numerical argument out of domain
seccomp_load_syscall_filter_set: add success
seccomp_load_syscall_filter_set: load success
Operating on architecture: x32
seccomp_load_syscall_filter_set: init success
Failed to add rule for system call cacheflush() / -10104, ignoring: Numerical argument out of domain
Failed to add rule for system call clock_getres_time64() / -10213, ignoring: Numerical argument out of domain
Failed to add rule for system call clock_gettime64() / -10214, ignoring: Numerical argument out of domain
Failed to add rule for system call clock_nanosleep_time64() / -10215, ignoring: Numerical argument out of domain
Failed to add rule for system call futex_time64() / -10222, ignoring: Numerical argument out of domain
Failed to add rule for system call get_thread_area() / 211, ignoring: Numerical argument out of domain
Failed to add rule for system call getegid32() / -10015, ignoring: Numerical argument out of domain
Failed to add rule for system call geteuid32() / -10016, ignoring: Numerical argument out of domain
Failed to add rule for system call getgid32() / -10017, ignoring: Numerical argument out of domain
Failed to add rule for system call getgroups32() / -10018, ignoring: Numerical argument out of domain
Failed to add rule for system call getresgid32() / -10019, ignoring: Numerical argument out of domain
Failed to add rule for system call getresuid32() / -10020, ignoring: Numerical argument out of domain
Failed to add rule for system call getuid32() / -10021, ignoring: Numerical argument out of domain
Failed to add rule for system call set_thread_area() / 205, ignoring: Numerical argument out of domain
Failed to add rule for system call set_tls() / -10183, ignoring: Numerical argument out of domain
Failed to add rule for system call sigreturn() / -10058, ignoring: Numerical argument out of domain
Failed to add rule for system call ugetrlimit() / -10068, ignoring: Numerical argument out of domain
seccomp_load_syscall_filter_set: add success
seccomp_load_syscall_filter_set: load success
Operating on architecture: x86-64
seccomp_load_syscall_filter_set: init success
Failed to add rule for system call cacheflush() / -10104, ignoring: Numerical argument out of domain
Failed to add rule for system call clock_getres_time64() / -10213, ignoring: Numerical argument out of domain
Failed to add rule for system call clock_gettime64() / -10214, ignoring: Numerical argument out of domain
Failed to add rule for system call clock_nanosleep_time64() / -10215, ignoring: Numerical argument out of domain
Failed to add rule for system call futex_time64() / -10222, ignoring: Numerical argument out of domain
Failed to add rule for system call getegid32() / -10015, ignoring: Numerical argument out of domain
Failed to add rule for system call geteuid32() / -10016, ignoring: Numerical argument out of domain
Failed to add rule for system call getgid32() / -10017, ignoring: Numerical argument out of domain
Failed to add rule for system call getgroups32() / -10018, ignoring: Numerical argument out of domain
Failed to add rule for system call getresgid32() / -10019, ignoring: Numerical argument out of domain
Failed to add rule for system call getresuid32() / -10020, ignoring: Numerical argument out of domain
Failed to add rule for system call getuid32() / -10021, ignoring: Numerical argument out of domain
Failed to add rule for system call set_tls() / -10183, ignoring: Numerical argument out of domain
Failed to add rule for system call sigreturn() / -10058, ignoring: Numerical argument out of domain
Failed to add rule for system call ugetrlimit() / -10068, ignoring: Numerical argument out of domain
seccomp_load_syscall_filter_set: add success
@default failed with exit status 1.
Assertion 'wait_for_terminate_and_check(syscall_filter_sets[i].name, pid, WAIT_LOG) == EXIT_SUCCESS' failed at src/test/test-seccomp.c:148, function test_filter_sets(). Aborting.
==23576== 
==23576== Process terminating with default action of signal 6 (SIGABRT): dumping core
==23576==    at 0x4C289D5: raise (in /usr/lib64/libc-2.32.so)
==23576==    by 0x4C118A3: abort (in /usr/lib64/libc-2.32.so)
==23576==    by 0x49D8296: log_assert_failed_realm (log.c:864)
==23576==    by 0x40307C: test_filter_sets (test-seccomp.c:148)
==23576==    by 0x409A89: main (test-seccomp.c:1037)
==23576== 
==23576== HEAP SUMMARY:
==23576==     in use at exit: 806 bytes in 30 blocks
==23576==   total heap usage: 46 allocs, 16 frees, 11,120 bytes allocated
==23576== 
==23576== 5 bytes in 1 blocks are possibly lost in loss record 1 of 25
==23576==    at 0x4839809: malloc (vg_replace_malloc.c:307)
==23576==    by 0x4C7B53E: strdup (in /usr/lib64/libc-2.32.so)
==23576==    by 0x58E6444: __trans_list_add (in /usr/lib64/libnl-3.so.200.26.0)
==23576==    by 0x586BECC: ??? (in /usr/lib64/libnl-route-3.so.200.26.0)
==23576==    by 0x40108ED: call_init.part.0 (in /usr/lib64/ld-2.32.so)
==23576==    by 0x40109D7: _dl_init (in /usr/lib64/ld-2.32.so)
==23576==    by 0x40010C9: ??? (in /usr/lib64/ld-2.32.so)
(snip)
```

Not sure, but this issue may be related to the recent Travis CI failures (but the failure is under ASAN). See, e.g., #17627.

### PR DESCRIPTION
**Pull request by yuwata**
Hopefully fixes #17630.

cc @mrc0mmand. The last commit which is for ASAN is not tested.

### CODE REVIEW

**Comment by mrc0mmand** [comment]
Can confirm that the ASan-related commit helps:

```
...
/* test_filter_sets */
Running on address sanitizer, skipping @default
...
@aio succeeded.
Running on address sanitizer, skipping @basic-io
Testing @chown
...
@setuid succeeded.
Running on address sanitizer, skipping @signal
Testing @swap
...
# echo $?
0
```

Thanks!

**Comment by keszybz** [comment]
semaphore:
```
Err:1 http://deb.debian.org/debian bullseye/main amd64 perl-modules-5.32 all 5.32.0-5
  Temporary failure resolving 'deb.debian.org'
```

File: src/test/test-seccomp.c [hunk 1]
```diff
@@ -323,6 +334,11 @@ static void test_protect_sysctl(void) {
 
                 assert_se(seccomp_protect_sysctl() >= 0);
 
+#if HAVE_VALGRIND_VALGRIND_H
+                if (!RUNNING_ON_VALGRIND)
```
Comment (keszybz) [inline src/test/test-seccomp.c]: I don't understand this: why is the test reversed here?

Comment (keszybz) [reply src/test/test-seccomp.c]: I assume this was a mistake. I force-pushed into your branch with this `if` reversed.

Comment (yuwata) [reply src/test/test-seccomp.c]: Oh, thank you!

**Pull request merged**